src/docfx/lib/markdown/MarkdownEngine.cs

```diff
@@ -319,35 +321,70 @@ private static ErrorBuilder GetErrors()
             return file is null ? default : (_input.ReadString(file).Replace("\r", ""), new SourceInfo(file));
         }
 
-        private string GetImageLink(string path, MarkdownObject origin, string? altText)
+        private string GetImageLink(string path, MarkdownObject origin, string? altText, bool? isIcon)
         {
             if (altText is null && origin is LinkInline linkInline && linkInline.IsImage)
             {
                 altText = ToPlainText(origin);
             }
-
-            return GetImageLink(new SourceInfo<string>(path, origin.GetSourceInfo()), origin, altText, -1);
+            var node = new ImageLinkNode
+            {
+                UrlLink = path,
+                ImageLinkType = isIcon == true ? ImageLinkType.Icon : ImageLinkType.Default,
+                AltText = altText,
+                IsInline = origin.IsInlineImage(-1),
+            };
+            return GetLink(new SourceInfo<string>(path, origin.GetSourceInfo()), origin, node);
         }
 
         private string GetImageLink(SourceInfo<string> href, MarkdownObject origin, string? altText, int imageIndex)
         {
-            _contentValidator.ValidateLink(GetRootFilePath(), href, origin, true, altText, imageIndex);
-            var link = GetLink(href, origin);
-            return link;
+            var node = new ImageLinkNode
+            {
+                UrlLink = href,
+                ImageLinkType = ImageLinkType.Default,
+                AltText = altText,
+                IsInline = origin.IsInlineImage(imageIndex),
+            };
+            return GetLink(href, origin, node);
         }
 
-        private string GetLink(SourceInfo<string> href, MarkdownObject origin)
+        private string GetLink(SourceInfo<string> link, MarkdownObject origin)
         {
-            _contentValidator.ValidateLink(GetRootFilePath(), href, origin, false, null, -1);
+            var node = new HyperLinkNode
+            {
+                UrlLink = link,
+                LinkText = (origin is LinkInline { IsImage: false }) ? ToPlainText(origin) : null,
```

Comment:
[This fix](https://github.com/dotnet/docfx/pull/7023/commits/b27159aa4cec3397beee5aab180230da7e747e43) of stack overflow seems a bit hack. We need a dedicated plain text markdown pipeline that does not do validation.

Reply:
Consider turn `IsImage: false` check into `Debug.Assert`